rust/relay/ebpf-shared/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+//! Shared data structures between the kernel and userspace.
+//!
+//! To learn more about the layout requirements of these structs, read <https://github.com/foniod/redbpf/issues/150#issuecomment-964017857>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct ClientAndChannelV4 {
+    ipv4_address: [u8; 4],
+    _padding_ipv4_address: [u8; 4],
+
+    port: u16,
+    _padding_port: [u8; 6],
+
+    channel: u16,
+    _padding_channel: [u8; 6],
+
+    _padding_struct: [u8; 40],
+}
```

Comment:
We force a C-layout here and perform manual padding. This is important because memory between the kernel and user-space is apparently copied in certain sizes, here 8 bytes. In order to not mess up our data, we need to manually align it to 8-byte boundaries.

Reply:
This was very painful to debug :joy: